### PR TITLE
document that destroy provisioners don't run with create_before_destroy

### PR DESCRIPTION
**Pull request by yvespp**
Documents issue https://github.com/hashicorp/terraform/issues/13549

Ran in this today again after first running into it a year ago and forgetting about it...
I think it's better to document this so people don't try to use this and then just end up disappointed because it doesn't work.

### CODE REVIEW

**Comment by yvespp** [comment]
`https://packer.io/` redirects to `https://www.packer.io/`. Should I update the link in the docs?

File: website/docs/language/meta-arguments/lifecycle.html.md [hunk 1]
```diff
@@ -45,6 +45,10 @@ The following arguments can be used within a `lifecycle` block:
     such features, so you must understand the constraints for each resource
     type before using `create_before_destroy` with it.
 
+    Destroy provisioners of this resource will not run if `create_before_destroy`
+    is used. This limitation may be addressed in the future, see
+    [GitHub issue](https://github.com/hashicorp/terraform/issues/13549) for details.
+
```
Comment (laurapacilio) [inline website/docs/language/meta-arguments/lifecycle.html.md:51]:
```suggestion
    Destroy provisioners of this resource will not run if `create_before_destroy`
    is set to `true`. We may address this in the future, and this [GitHub issue](https://github.com/hashicorp/terraform/issues/13549) contains more details.

```

File: website/docs/language/resources/provisioners/syntax.html.md [hunk 2]
```diff
@@ -237,6 +237,11 @@ fail, Terraform will error and rerun the provisioners again on the next
 `terraform apply`. Due to this behavior, care should be taken for destroy
 provisioners to be safe to run multiple times.
 
+Destroy provisioners will not run if the lifecycle Meta-Argument
+[`create_before_destroy`](/docs/language/meta-arguments/lifecycle.html) is used
+in the resource. This limitation may be addressed in the future, see
+[GitHub issue](https://github.com/hashicorp/terraform/issues/13549) for details.
```
Comment (laurapacilio) [inline website/docs/language/resources/provisioners/syntax.html.md:243]:
```suggestion
    Destroy provisioners of this resource will not run if `create_before_destroy`
    is set to `true`. We may address this in the future, and this [GitHub issue](https://github.com/hashicorp/terraform/issues/13549) contains more details.
```

**Review by laurapacilio — COMMENTED** [review]
Thanks for adding this - We appreciate you making the effort to document this, as it's clearly an issue that folks care deeply about! I made a few edits just to comply with our style guide - see what you think!

**Comment by yvespp** [comment]
Thanks for the review!

**Comment by laurapacilio** [comment]
Merging even though link checker is failing because the Packer website link isn't really broken :-)

**Pull request merged**